### ISSUE TITLE
Route Params not detected by Angular Universal on Server Side

### ISSUE THREAD

**Issue by nikhilbhalwankar** [angular/universal#1958]
# 🐞 Bug report

#### What modules are related to this issue?

- [ ] aspnetcore-engine
- [ ] builders
- [ ] common
- [*] express-engine
- [ ] hapi-engine

### Description
Route Params are not detected by Angular Universal on Server Side. Looks like they get detected when code execution happens on client side.

## 🔬 Minimal Reproduction
Below code snippet gives error - **TypeError: Cannot read property ‘userId’ of undefined**. Meta tag does not get rendered on web page. However, the console.log() statement correctly prints **pId**.

<pre><code>

this.route.params.subscribe(params => {

        this.paramPostId = params["pId"];
        this.angFire.doc('featuredPosts/' + this.paramPostId).valueChanges().subscribe(userData => {
          let uPost: UserPost = userData as UserPost;
          console.log(this.paramPostId);
          this.metaTagService.updateTag(
            { property: 'og:title', content: 'Post of ' + uPost.userId }
          );
        });
      
    });
</code></pre>

------

Below code works perfectly fine and meta tag is also rendered on web page source. Instead of taking pId from route param, I tried adding it manually.

<pre><code>

this.paramPostId='ABCDSFTZRZ16EA5EF';
this.angFire.doc('featuredPosts/' + this.paramPostId).valueChanges().subscribe(userData => {
   let uPost: UserPost = userData as UserPost;
   this.metaTagService.updateTag({ property: 'og:title', content: 'Post of ' + uPost.userId });
});
</code></pre>

## 🔥 Exception or Error
<pre><code>
TypeError: Cannot read property ‘userId’ of undefined
</code></pre>

## 🌍 Your Environment
<pre><code>
Angular CLI: 11.0.5
Node: 12.18.0
OS: linux x64

Angular: 11.0.5
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... platform-server, router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1100.5
@angular-devkit/build-angular   0.1100.5
@angular-devkit/core            11.0.5
@angular-devkit/schematics      11.0.5
@angular/cdk                    11.0.3
@angular/fire                   6.1.4
@angular/material               11.0.3
@nguniversal/builders           11.0.1
@nguniversal/express-engine     11.0.1
@schematics/angular             11.0.5
@schematics/update              0.1100.5
rxjs                            6.6.3
typescript                      4.0.5
</code></pre>

Basically when the **pId** is hard coded then the call to cloud firestore database works fine. No error is thrown by Angular Universal. Meta tag for SEO is also getting generated on web page. But when the **pId** is read from **route params**, reading the route params fails on server side and undefined error is thrown.

**Comment by alan-agius4** [comment]
I'm sorry, but we can't reproduce the problem following the instructions you provided.


If the problem persists, please open a new issue following [our submission guidelines](https://github.com/angular/universal/blob/master/CONTRIBUTING.md#-submitting-an-issue).

A good way to make a minimal repro is to create a new app via `ng new repro-app` and add the minimum possible code to show the problem. Then you can push this repository to github and link it here.
